libs/stack/stack-ui/src/components/fields/Select/components/useListboxSections.ts

```
@@ -0,0 +1,61 @@
+        return {
+          sections: updatedSections,
+          currentSection: { header: item, items: [] },
+          orphanedItems: acc.orphanedItems,
+        }
+      }
+
+      // For non-header items
+      if (acc.currentSection) {
+        // Add to current section
+        return {
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)